server/container/mailbox-adapter/src/main/java/org/apache/james/adapter/mailbox/MailReprocessingManagement.java

```diff
@@ -0,0 +1,58 @@
+package org.apache.james.adapter.mailbox;
```

Comment:
Idem, you miss the license

Reply:
I do believe that such a class should not be in `mailbox-adapter` because none of the two components are related to mailbox.

What about creating a `server/container/cli-components-api` maven module (for the MBean, cli can depend on it) and a `server/container/cli-components` (for the implementation, server would depend on it.)

It looks to some work, still to be contributed, that I did recently.

Reply:
@chibenwa I already know that residing the class in mailbox wasn't proper. I just wanted to get feedback from you about where it should be resided. That's why i pushed them premature.

Reply:
@chibenwa i have already implemented all the functionalities described in James-1050, also required to review :) but before pushing them all, i will work on your suggestion described here.